galsim/gsobject.py

```diff
@@ -1677,17 +1674,24 @@ def drawImage(self, image=None, nx=None, ny=None, bounds=None, scale=None, wcs=N
                 raise GalSimIncompatibleValuesError(
                     "poisson_flux is only relevant for method='phot'",
                     method=method, sensor=sensor, poisson_flux=poisson_flux)
-            if photon_ops != ():
+
+        # If using photon ops with fft, then need a sensor.
+        if photon_ops != () and method != 'phot' and sensor is None:
```

Comment:
Maybe `tuple(photon_ops) != ()`, or some other way of checking for a generic empty Sequence?

Reply:
I'm not sure.  
`()` is the default value if the user doesn't specify something else, which is why that's what I'm comparing here.  If the user specifies `[]` for instance instead, that feels intentional, like maybe the photon_ops was generated programmatically, and happened not to have any elements after whatever conditions were run through.  In that case, I think the user might hit some of these sanity checks, thinking it was ok to pass `rng`, since some of the potential photon_ops that might have been added (but weren't) need it.  The pure default empty tuple seems most likely to be just the default value, not an accidental user-supplied empty tuple.

Reply:
Sure, it just feels weird to treat `[]` and `()` differently when whether the Sequence is mutable or not really doesn't matter here.  Could make the default `None` I guess.  Or just leave it alone; I'm sure this is a pretty obscure edge case.

Reply:
I agree.  I made the default value None.  That's more in line with how we do other defaults, so that's probably the better design here.